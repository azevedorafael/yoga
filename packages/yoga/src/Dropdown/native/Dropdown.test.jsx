import React from 'react';
import { render } from '@testing-library/react-native';

import { ThemeProvider, Dropdown } from '../..';

describe('<Dropdown />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Dropdown />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
