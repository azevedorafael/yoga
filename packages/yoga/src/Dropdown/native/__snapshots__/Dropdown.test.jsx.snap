// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Snapshots should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="button"
  >
    <View
      disabled={false}
      full={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "position": "relative",
            "width": 320,
          },
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#9898A6",
            },
          ]
        }
      >
        Find an activity to love
      </Text>
      <RNSVGSvgView
        fill="#9898A6"
        focusable={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            Object {
              "opacity": 1,
            },
            null,
          ]
        }
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4288190630,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot when disabled 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="button"
  >
    <View
      disabled={true}
      full={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBDBE8",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "position": "relative",
            "width": 320,
          },
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#DBDBE8",
            },
          ]
        }
      >
        Find an activity to love
      </Text>
      <RNSVGSvgView
        fill="#DBDBE8"
        focusable={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            Object {
              "opacity": 1,
            },
            null,
          ]
        }
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4292598760,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot when full 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="button"
  >
    <View
      disabled={false}
      full={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D3D3E2",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "position": "relative",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#9898A6",
            },
          ]
        }
      >
        Find an activity to love
      </Text>
      <RNSVGSvgView
        fill="#9898A6"
        focusable={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            Object {
              "opacity": 1,
            },
            null,
          ]
        }
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4288190630,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot when has a selected value 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableWithoutFeedback
    accessibilityRole="button"
  >
    <View
      disabled={true}
      full={false}
      selected={
        Object {
          "label": "Swimming",
          "selected": true,
          "value": "swimming",
        }
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#41414A",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "position": "relative",
            "width": 320,
          },
        ]
      }
    >
      <Text
        disabled={true}
        selected={
          Object {
            "label": "Swimming",
            "selected": true,
            "value": "swimming",
          }
        }
        style={
          Array [
            Object {
              "color": "#41414A",
            },
          ]
        }
      >
        Swimming
      </Text>
      <RNSVGSvgView
        fill="#DBDBE8"
        focusable={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            Object {
              "opacity": 1,
            },
            null,
          ]
        }
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4292598760,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
  </TouchableWithoutFeedback>
</View>
`;
